receiver/couchdbreceiver/metrics.go

```diff
@@ -0,0 +1,199 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package couchdbreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/couchdbreceiver"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/receiver/scrapererror"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/couchdbreceiver/internal/metadata"
+)
+
+func (c *couchdbScraper) recordCouchdbAverageRequestTimeDataPoint(now pdata.Timestamp, stats map[string]interface{}, errors scrapererror.ScrapeErrors) {
+	averageRequestTimeMetricKey := []string{"request_time", "value", "arithmetic_mean"}
+	averageRequestTimeValue, err := getValueFromBody(averageRequestTimeMetricKey, stats)
+	if err != nil {
+		errors.AddPartial(1, err)
+		return
+	}
+
+	parsedValue, err := c.parseFloat(averageRequestTimeValue)
+	if err != nil {
+		errors.AddPartial(1, err)
+		return
+	}
+	c.mb.RecordCouchdbAverageRequestTimeDataPoint(now, parsedValue)
+}
+
+func (c *couchdbScraper) recordCouchdbHttpdBulkRequestsDataPoint(now pdata.Timestamp, stats map[string]interface{}, errors scrapererror.ScrapeErrors) {
+	httpdBulkRequestsMetricKey := []string{"httpd", "bulk_requests", "value"}
+	httpdBulkRequestsMetricValue, err := getValueFromBody(httpdBulkRequestsMetricKey, stats)
+	if err != nil {
+		errors.AddPartial(1, err)
+		return
+	}
+
+	parsedValue, err := c.parseInt(httpdBulkRequestsMetricValue)
+	if err != nil {
+		errors.AddPartial(1, err)
+		return
+	}
+	c.mb.RecordCouchdbHttpdBulkRequestsDataPoint(now, parsedValue)
+}
+
+func (c *couchdbScraper) recordCouchdbHttpdRequestsDataPoint(now pdata.Timestamp, stats map[string]interface{}, errors scrapererror.ScrapeErrors) {
+	methods := []string{metadata.AttributeHTTPMethod.COPY, metadata.AttributeHTTPMethod.DELETE, metadata.AttributeHTTPMethod.GET, metadata.AttributeHTTPMethod.HEAD, metadata.AttributeHTTPMethod.OPTIONS, metadata.AttributeHTTPMethod.POST, metadata.AttributeHTTPMethod.PUT}
```

Comment:
Same question as the status codes. Just making sure - no false errors here?

Reply:
Same. the json response will return 0 values for each response method. It will not be omitted.

Reply:
Thanks for confirming.